### PR TITLE
Updates for pvlib v0.7

### LINKED ISSUE #44: Update to work with pvlib v0.7
There are a few changes introduced in pvlib 0.7 that introduced errors in pvcaptest.  

The Travis CI requirements file did not have the pvlib version pinned prior to discovering the issues with pvlib 0.7 when Travis ran for name changing pull request #43.  I temporarily pinned pvlib to 0.6.3 to avoid errors in #43.

I would like to create a separate branch to address updates for pvlib compatibility.

Looks like the failure is caused by the name changes in the SAM inverter file. Change [this ](https://github.com/pvcaptest/pvcaptest/blob/4df1283d336d2cd7009fc51501e58cfe8795b191/captest/capdata.py#L657) line to

`    cec_inverter = cec_inverters['ABB__MICRO_0_25_I_OUTD_US_208__208V_']`

I think pvlib v0.7 will work for you then.

_Originally posted by @cwhanse in https://github.com/pvcaptest/pvcaptest/pull/43_

### PR DESCRIPTION
**Pull request by cwhanse**
closes #44. Let's see what else fails...

### CODE REVIEW

**Comment by cwhanse** [comment]
@bt- the other test failures all have the same cause. The `pvlib.ModelChain.prepare_inputs` method now requires the user to provide weather data; before, if weather was omitted, pvlib ran a clearsky model for you (and didn't really tell you that's what it was doing).

**Comment by cwhanse** [comment]
@bt- I think this covers the pvlib 0.7 migration. It was a good exercise for me to go back to pvlib's release notes and see if I could figure out what had changed.

File: tests/test_CapData.py [hunk 1]
```diff
@@ -873,7 +874,7 @@ def test_csky_not_concat_all(self):
         self.assertEqual(csky_ghi_poa.shape[1], 8,
                          'Returned dataframe does not have 5 columns.')
```
Comment (bt-) [inline tests/test_CapData.py]: I noticed a copy paste error here - the 5 in the text string should be an 8.

**Comment by cwhanse** [comment]
> the actual module and inverter is not important.

captest is creating the `PVSystem` instance to get access to solar position and clear sky methods, is that correct? If so, then yes I think `[:, 0]` is OK.  The names in that data set are outside of pvlib's control, so we have to adapt as they change. We hardwired the parameters for a module and inverter into our tests.

**Comment by bt-** [comment]
Yes, that is correct- captest is currently creating the `PVSystem` instance just to gain access to the solar position and clear sky methods.  If I remember correctly, I decided it would be easier to use the methods than accessing pvlib functions directly.  Also, I thought creating the `PVSystem` instance would be helpful if somewhere far down the road pvcaptest was able to run a pvlib model.  It would potentially be nice to have the modelling in the same workflow as the capacity test.

**Comment by cwhanse** [comment]
That design makes sense to me. I just wanted to confirm that the module/inverter parameters aren't being used elsewhere. An alternative would be to specify the PVWatts model and hardwire it's parameters into this function.

**Comment by bt-** [comment]
Thanks, that should help prevent that particular issue from cropping up again.  The last change needed, hopefully, is the pvlib version required in setup.py.  There are two references there that should be updated to `pvlib>=0.7.0`:

In the 'csky' extras
https://github.com/pvcaptest/pvcaptest/blob/4df1283d336d2cd7009fc51501e58cfe8795b191/setup.py#L33
and just below in the 'all' extras
https://github.com/pvcaptest/pvcaptest/blob/4df1283d336d2cd7009fc51501e58cfe8795b191/setup.py#L36

I'm trying to decide if the version of pvlib in the yml files used by Travis to create the test environments should be pinned to 0.7.0 as well.  I think leaving it unpinned should be fine for now.

**Comment by cwhanse** [comment]
> Thanks, that should help prevent that particular issue from cropping up again.  The last change needed, hopefully, is the pvlib version required in setup.py.  There are two references there that should be updated to pvlib>=0.7.0:
> In the 'csky' extras
> 
>   
>     
>       pvcaptest/setup.py
>     
>     
>          Line 33
>       in
>       4df1283
>     
>     
>     
>     
> 
>         
>           
>            'csky': ['pvlib>0.6', 
>         
>     
>   
> 
> 
> and just below in the 'all' extras
> 
>   
>     
>       pvcaptest/setup.py
>     
>     
>          Line 36
>       in
>       4df1283
>     
>     
>     
>     
> 
>         
>           
>            'pvlib>0.6', 
>         
>     
>   
> 
> 
> I'm trying to decide if the version of pvlib in the yml files used by Travis to create the test environments should be pinned to 0.7.0 as well.  I think leaving it unpinned should be fine for now.

Unpinned may be better, to detect failures for future releases of pvlib. We're already moving on 0.7.2, and expect 0.8 in the next few months. I don't think these releases affect the functions used by pvcaptest.

**Pull request merged**